### DISCUSSION TITLE
Github Actions url to run

### DISCUSSION THREAD

**Discussion by judoole** [community/community#25470] [Actions]
<p>Hi!</p>
<p>I’m trying to create a workflow for deployment, and I’d like to <a href="https://developer.github.com/v3/repos/deployments/#create-a-deployment-status" rel="nofollow noopener">create a deploy status</a> and submit the url to the log of the action that deployed. Very handy if the deply failes I think. Using <a href="https://github.com/actions/bin/tree/master/debug" rel="nofollow noopener">actions/bin/debug</a> I can’t seem to find any reference to the id of the run or url to the run.</p>
<p>Is there a way to extract this in some way?</p>

**Comment by mcolyer** [comment]
<p>We don’t currently expose that information, however I’ve made a note of this feedback so we can take it into account as we continue to improve actions.</p>

**Comment by judoole** [comment]
<p>Thanks! It would perhaps be great to have it as one of the <a href="https://developer.github.com/actions/creating-github-actions/accessing-the-runtime-environment/#environment-variables" rel="nofollow noopener">environment variables</a>. GITHUB_RUN_ID or something along those lines.</p>

**Comment by podviaznikov** [comment]
<p>I have similar use case. Would be nice to have this.</p>

**Comment by amirrustam** [comment]
<p>Any updates on this now with the launch of CI/CD support? </p>
<p>Exposing the run ID would be really helpful for supporting parallelization for <a href="https://cypress.io" rel="nofollow noopener">Cypress</a> users.</p>

**Comment by colinjfw** [comment]
<p>I was just trying to solve this problem recently, this is what I ended up with in the mean time to just set the deployment url:</p>
<p><a href="https://github.com/deliverybot/status/blob/master/index.js#L7" rel="nofollow noopener">https://github.com/deliverybot/status/blob/master/index.js#L7</a></p>

**Comment by davidjdixon** [comment]
<p><a class="mention" href="/mcolyer">@mcolyer</a> has there been any movement on this item?</p>

**Comment by ivailop** [comment]
<p>As requested in <a href="https://github.community/t/check-suite-identifier-as-an-environment-variable/16642">https://github.community/t5/GitHub-Actions/Check-Suite-identifier-as-an-environment-variable/m-p/35498#M2240</a>, having GITHUB_CHECK_SUITE_ID would allow provision of such URL as follow:</p>
<pre><code>GITHUB_WORKFLOW_URL=https://github.com/$GITHUB_REPOSITORY/commit/$GITHUB_SHA/checks?check_suite_id=$GITHUB_CHECK_SUITE_ID
</code></pre>

**Comment by ob1dev** [comment]
<p>Hi <a class="mention" href="/mcolyer">@mcolyer</a> </p>
<p>Another example would be, if I want to send a Slack message with a link to my GitHub Actions run in case if it sucessed, failed or canceled, I want a varible that points to:</p>
<pre><code>https://github.com/{account-name}/{repository-name}/commit/{git-sha}/checks?check_suite_id={run-id}
</code></pre>
<p>Best Wishes,</p>
<ul>
<li>Oleg</li>
</ul>

**Comment by ivailop** [comment, marked as answer]
<p>With the new <a href="https://developer.github.com/v3/actions/" rel="nofollow noopener">Actions API</a>, there are now $GITHUB_RUN_ID environment variable provided. </p>
<p>Using it, a URL to the workflow could be constructed as follows: </p>
<pre><code>GITHUB_WORKFLOW_URL=https://github.com/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID
</code></pre>

**Reply by anden-dev** [reply]
Universal ( with GHE Support)

```sh
GITHUB_WORKFLOW_URL=$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID
```

**Reply by tpak** [reply]
I assume that @anden-dev's answer works if you are using the API, I just happened to stumble here looking for the same thing and took what @anden-dev  proposed and modified it to work in an action job step and it works great. I am able to substitute that string in other steps as needed, I just used echo to demonstrate so anyone can test it. 

This works in a workflow action without getting involved with the Actions API
```
      - name: Echo GHA url 
        run: echo ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
        shell: bash
```

**Reply by mloskot** [reply]
@anden-dev  Although it is indeed documented at the bottom of the [Default environment variables](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables), FWIW, this technique does not work in the [entire workflow `env:` section](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#env). Instead, the one presented by @tpak works:

```yaml
env:
  BUILD_URL: "${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
```

**Reply by tpak** [reply]
@mloskot - indeed that is how we are using it in our actions to expose it across steps if necessary (in one sending it to the Slack notify step)

**Comment by harveyr** [comment]
<p>Does that actually work for you? My run IDs are different than the numbers in the URLs. And using the run ID like that results in a 404.</p>
<p>Edit: Aha! I’m wrong. There’s <code>\&lt;repo\&gt;/actions/runs/\&lt;id\&gt;</code> that shows all the jobs and <code>\&lt;repo\&gt;/runs/\&lt;some-other-number\&gt;</code> once you start clicking into jobs. The first one works as you say. My mistake!</p>

**Comment by harveyr** [comment]
<p>Formatting got weird there—sorry.</p>

**Comment by samuela** [comment]
<p>Does anyone know how to get that URL for the current job in the current run? <code>GITHUB_RUN_NUMBER</code> looks promising but doesn’t seem to be related to the URLs at all.</p>

**Comment by dominikfoldi** [comment]
<p>Yeah! It would be great to have the id of the current job to be able to construct a link for that too! Currently, this is not possible.</p>

**Comment by judoole** [comment]
<p>I think this should do it <a class="mention" href="/dominikfoldi">@dominikfoldi</a> <a class="mention" href="/samuela">@samuela</a><br>
<a href="https://github.com/%24GITHUB_REPOSITORY/actions/runs/%24GITHUB_RUN_ID" class="onebox" target="_blank" rel="noopener nofollow ugc">https://github.com/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID</a></p>
<p>We are currently using it in our action for updating the deployment status <a href="https://github.com/unacast/actions-github-deployment-status/blob/d6be60b7d328f52d8a329119f39add23737be3da/bin/deployment-create-status#L14" rel="noopener nofollow ugc">https://github.com/unacast/actions-github-deployment-status/blob/d6be60b7d328f52d8a329119f39add23737be3da/bin/deployment-create-status#L14</a></p>

**Comment by dominikfoldi** [comment]
<p>No. That will link to the workflow run which has one or more jobs. The jobs have their own dedicated ids, so they have their own links, too.</p>

**Comment by judoole** [comment]
<p>Ah, sorry. Then I understand. And I don’t know.</p>

**Comment by Simran-B** [comment]
<p>The job run IDs don’t seem to be exposed in actions, at least per the documentation. Might be worth to print all contexts etc. to check whether this information is available but undocumented.</p>
<p>Other than that, you may use the API to get the job <code>id</code>s:</p>
<pre><code>https://api.github.com/repos/{owner}/{repo}/actions/runs/{run_id}/jobs
</code></pre>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#list-jobs-for-a-workflow-run" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="16" height="16">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#list-jobs-for-a-workflow-run" target="_blank" rel="noopener nofollow ugc">Actions - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by justinmchase** [comment]
This shows an existing run, is it possible to create a hyperlink that will create a new run when clicked?
